app/test/vesting_test.go

```diff
@@ -0,0 +1,550 @@
+	assert.EqualValues(s.T(),
+		testfactory.BaseAccountDefaultBalance,
+		qAcc.GetOriginalVesting().AmountOf(app.BondDenom).Int64(),
+		"original vesting must match")
+
+	/*--------*/
+
+	// Test available balance
+	for i := 0; i < 5; i++ {
+		queryTime := tmtime.Now()
+
+		balances, err := GetAccountSpendableBalance(cctx.GRPCClient, accAddress)
+		assert.NoError(s.T(), err)
+
+		minExpectedSpendableBal := cva.GetVestedCoins(queryTime).Add(initBalanceForGasFee)
+
+		assert.LessOrEqual(s.T(),
+			balances.AmountOf(app.BondDenom).Int64(),
+			minExpectedSpendableBal.AmountOf(app.BondDenom).Int64(),
+		)
+
+		_, err = cctx.WaitForTimestamp(startTime.Add(5)) // Wait for a while
+		assert.NoError(s.T(), err)
+	}
+	/*--------*/
+
+	s.testTransferVestingAmount(cctx, accName, false)
+	s.testDelegation(cctx, accName)
+	s.testClaimDelegationReward(cctx, accName)
+}
+
+func (s *VestingModuleTestSuite) testTransferVestingAmount(cctx testnode.Context, accName string, mustSucceed bool) {
+	randomAcc := s.unusedAccount()
+	msgSend := banktypes.NewMsgSend(
+		getAddress(accName, cctx.Keyring),
+		getAddress(randomAcc, cctx.Keyring),
+		sdk.NewCoins(sdk.NewCoin(app.BondDenom, sdk.NewInt(testfactory.BaseAccountDefaultBalance))), // try to transfer the locked amount
+	)
+	encCfg := encoding.MakeConfig(app.ModuleEncodingRegisters...)
+	resTx, err := testnode.SignAndBroadcastTx(encCfg, cctx.Context, accName, []sdk.Msg{msgSend}...)
+	assert.NoError(s.T(), err)
+
+	resQ, err := cctx.WaitForTx(resTx.TxHash, 10)
+	assert.NoError(s.T(), err)
+
+	if mustSucceed {
+		assert.EqualValues(s.T(), 0, resQ.TxResult.Code, "the transfer TX must succeed")
+	} else {
+		assert.EqualValues(s.T(), 5, resQ.TxResult.Code, "the transfer TX must fail")
+	}
+}
+
+func (s *VestingModuleTestSuite) testDelegation(cctx testnode.Context, accName string) {
+	accAddress := getAddress(accName, cctx.Keyring).String()
+
+	del, err := GetAccountDelegations(cctx.GRPCClient, accAddress)
+	assert.NoError(s.T(), err)
+	assert.Empty(s.T(), del, "initial delegation must be empty")
+
+	validators, err := GetValidators(cctx.GRPCClient)
+	assert.NoError(s.T(), err)
+	assert.NotEmpty(s.T(), validators)
+
+	msgDelg := stakingtypes.NewMsgDelegate(
+		getAddress(accName, cctx.Keyring),
+		validators[0].GetOperator(),
+		sdk.NewCoin(app.BondDenom, sdk.NewInt(testfactory.BaseAccountDefaultBalance)),
+	)
+	encCfg := encoding.MakeConfig(app.ModuleEncodingRegisters...)
+	resTx, err := testnode.SignAndBroadcastTx(encCfg, cctx.Context, accName, []sdk.Msg{msgDelg}...)
+	assert.NoError(s.T(), err)
+
+	resQ, err := cctx.WaitForTx(resTx.TxHash, 10)
+	assert.NoError(s.T(), err)
+	assert.EqualValues(s.T(), 0, resQ.TxResult.Code, "the delegation TX must succeed")
+
+	// verify the delegations
+	del, err = GetAccountDelegations(cctx.GRPCClient, accAddress)
+	assert.NoError(s.T(), err)
+	assert.NotEmpty(s.T(), del, "delegations must not be empty")
+	assert.EqualValues(s.T(),
+		testfactory.BaseAccountDefaultBalance,
```

Comment:
[question] it's not immediately clear to me why the base account default balance should match the delegation amount.

Would it be helpful if this test file described it's own constant named `DelegationAmount` and used it here?

Reply:
Since some accounts like Periodic and Continues can have part of their balance locked (vesting), and it is hard to tell in advance how much, to test if locked balance can be delegated, the total initial vesting amount was set to delegate to cover it.
we need to explain it in comments